container.go

```diff
@@ -79,6 +79,9 @@ type ContainerConfig struct {
 	Annotations map[string]string
 
 	Mounts []Mount
+
+	// Devices are devices that must be available within the container.
+	Devices []Device
```

Comment:
Store the `Device` structure won't go well since it is an interface. I mean that marshal it into a json will work but if we want to retrieve it, we won't be able to determine what kind of device implementation is behind this. This should be another type here, with all the infos we need to rebuild this structure from scratch. I think this is something we're gonna need in order to detach properly the device.

Reply:
Yes, maybe we will need to store a device type constant/string explicitly for the device for unmarshalling. Again, would like to tackle the unmarshalling in my next PR, where it is actually needed to store devices to detach them.